crates/wasmtime/src/runtime/vm/const_expr.rs

```diff
@@ -118,7 +118,11 @@ impl<'a> ConstEvalContext<'a> {
             .borrow(shared_ty)
             .expect("should have a registered type for struct");
         let WasmSubType {
-            composite_type: WasmCompositeType::Struct(struct_ty),
+            composite_type:
+                WasmCompositeType {
+                    shared: false, // TODO: handle shared
```

Comment:
I think this is correct to have, this is basically pattern-matching to only unshared structures for `struct.new_default` which is desired behavior at this time. This panic can then get weeded out in the future once support for `struct.new_default` comes around.